### PR TITLE
Implement `elseif` blocks with macro

### LINKED ISSUE #96: taylorize macro does not recognize `elseif` statements
In current master, when trying to use `@taylorize` with the following function definition

```julia
@taylorize function f!(dq, q, params, t)
    for i in 1:10
        if i == 1
            dq[i] = 2q[i]
        elseif i == 2
            dq[i] = -q[i]
        else
            dq[i] = q[i]
        end
    end
    nothing
end
```

An error occurs:

```
ERROR: LoadError: ArgumentError: collection must be non-empty
```

If, instead, the `elseif` statement is removed, things work fine:
```julia
@taylorize function f!(dq, q, params, t)
    for i in 1:10
        if i == 1
            dq[i] = 2q[i]
        else
            dq[i] = q[i]
        end
    end
    nothing
end
```

### PR DESCRIPTION
**Pull request by lbenet**
Fixes #96

### CODE REVIEW

**Comment by lbenet** [comment]
I've included `continue` and `break` statements as well.

The solution does not shine because its elegance, but I think it works.

**Comment by lbenet** [comment]
I still need to provide some tests for the new cases...

**Comment by lbenet** [comment]
@PerezHz I have added some tests, in what I believe is the *expected* way the macro acts. Can you please take a look on those, especially in the parts in which not-recognized functions are just copied?

File: src/parse_eqs.jl [hunk 1]
```diff
@@ -340,9 +340,17 @@ function _newfnbody(fnbody, fnargs, d_indx)
...
                 append!(v_arraydecl, tmp_arraydecl)
             elseif ex_head == :if
+                # The first argument of an `if` expression is the condition, the
+                # second one is the block the condition is true, and the third
+                # is the `else`-block or an `ifelse`-block.
                 push!(newfnbody.args, Expr(:if, ex.args[1]))
                 for exx in ex.args[2:end]
-                    ifbody, tmp_newindx, tmp_arraydecl = _newfnbody( exx, fnargs, d_indx)
+                    if exx.head == :elseif
+                        exxx = Expr(:block, Expr(:if, exx.args[1].args[2], exx.args[2:end]...))
```
Comment (PerezHz) [inline src/parse_eqs.jl]: Nice workaround! I still can't figure out why `:elseif` blocks can't be treated straight-forward as `:if` blocks 🤷‍♂

File: test/taylorize.jl [hunk 2]
```diff
@@ -129,7 +143,7 @@ using Elliptic
 
     # Pendulum integrtf = 100.0
     @testset "Integration of the pendulum and DiffEqs interface" begin
-        @taylorize function pendulum!(dx, x, p, t)
+        @taylorize function pendulum!(dx::Array{T,1}, x::Array{T,1}, p, t) where {T}
```
Comment (PerezHz) [inline test/taylorize.jl]: Are we just testing type annotations here in the function definition, is there another reason to add them?

Comment (lbenet) [reply test/taylorize.jl]: Yes, you are right. Internally, we simply ignore the type annotations, but there was no test.

**Comment by lbenet** [comment]
Thanks @PerezHz for the review. As far as I am concerned, I think this is ready for merging, specially if it works fine for you. So, please, go ahead and merge this.

**Comment by PerezHz** [comment]
Merging! Again, thanks a lot @lbenet for taking care of this!

**Pull request merged**